containerfixture/containerfixture.go

```diff
@@ -0,0 +1,118 @@
+package containerfixture
+
+import (
+	_ "embed"
+	"fmt"
+	"log"
+	"math/rand"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+//go:embed containerfixture.yaml
+var manifest string
+
+const Username = "admin"
+const Password = "Password"
+
+// Generates a random dynamic port number
+func GetRandomPort() string {
+	minPort := 49152
+	maxPort := 65535
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	randomPort := fmt.Sprintf("%d", r.Intn(maxPort-minPort)+minPort)
+	return randomPort
+}
+
+// Modify the manifest to allow running it in parallel as
+// many times as needed with dynamic values
+func modifyYaml(yaml string) map[string]string {
+	podName := fmt.Sprintf("test-pod-%s", GetRandomPort())
+	webPort := GetRandomPort()
+	apiPort := GetRandomPort()
+
```

Comment:
perhaps we could use go templates here instead of simply replacing fixed strings - we could add a custom template function for getting a random port number while also storing it in a backing map.

Reply:
I'll check it out

Reply:
Applied simple templating, will see to the backing map